docs/architecture/adr-030-msg-auth-module.md

```diff
@@ -0,0 +1,218 @@
+# ADR 030: Msg Authorization Module
+
+## Changelog
+
+- 2019-11-06: Initial Draft
+- 2020-08-18: Updated Draft
+
+## Status
+
+Accepted
+
+## Context
+
+
+## Decision
+
+We will create a module named `msg_authorization`. The `msg_authorization` module provides support for
+granting arbitrary capabilities from one account (the granter) to another account (the grantee). Capabilities
+must be granted for a particular type of `sdk.Msg` one by one using an implementation
+of `Capability`.
+
+### Types
+
+Capabilities determine exactly what action is delegated. They are extensible
+and can be defined for any sdk.Msg type even outside of the module where the Msg is defined.
+
+#### Capability
+
+```go
+type Capability interface {
+	// MsgType returns the type of Msg's that this capability can accept
+	MsgType() sdk.Msg
```

Comment:
@aaronc , can this `MsgType()` be string directly? I don't see any other usages and we are just using msg.(type) .

Reply:
Yes, it can and it should be the return value of `proto.MessageName`. Although with #7458, we may switch to the fully-qualified service method name.

Reply:
Actually now it's updated to reflect ADR 031.